lib/helpers.js

```diff
@@ -25,7 +27,13 @@ const getFileType = (fileName, { transformExtensions }) => {
 };
 
 const reduceAssets = (files, asset, moduleAssets) => {
-  const name = moduleAssets[asset.name] ? moduleAssets[asset.name] : asset.info.sourceFilename;
+  let name;
+  if (moduleAssets[asset.name]) {
+    name = moduleAssets[asset.name];
+  } else if (asset.info.sourceFilename) {
+    name = join(dirname(asset.name), basename(asset.info.sourceFilename));
```

Comment:
This is basically the same thing that's already being done at the bottom of `hooks.js` for the `file-loader`/ NormalModule situation - https://github.com/shellscape/webpack-manifest-plugin/blob/9f408f609d9b1af255491036b6fc127777ee6f9a/lib/hooks.js#L122